pkg/rules/rules.go

```diff
@@ -51,6 +51,35 @@ func (r RuleNamespace) LintPromQLExpressions() (int, int, error) {
 	return count, mod, nil
 }
 
+// CheckRecordingRules checks that recording rules have at least one colon in their name, this is based
+// on the recording rules best practices here: https://prometheus.io/docs/practices/rules/
+// Returns the number of rules that don't match the requirements, and error if that number is not 0.
+func (r RuleNamespace) CheckRecordingRules(strict bool) int {
+	var name string
+	var count int
+	reqChunks := 2
+	if strict {
+		reqChunks = 3
+	}
+	for _, group := range r.Groups {
+		for _, rule := range group.Rules {
+			name = rule.Record
```

Comment:
recording rules and alerts could be defined in the same file and they use the same `Rules` key in the Prometheus file format. This will throw an error for all the alerts, as their `rule.Record` will be the nil value.